c2corg_api/views/document.py

```diff
@@ -336,6 +338,40 @@ def _check_update_type(self, document, old_versions):
                 'trying do update the document with the same content')
         return (update_types, changed_langs)
 
+    def _get_version(self, clazz, schema, adapt_schema=None):
+        id = self.request.validated['id']
+        lang = self.request.validated['lang']
+        version_id = self.request.validated['version_id']
+        document = self._get_in_lang(
```

Comment:
@tsauerwein are you sure the return document is the one form the requested version?
Shouldn't it be looked in the archives?